src/poolman.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2015 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <algorithm>
+#include <vector>
+#include "amount.h"
+#include "main.h"
+#include "poolman.h"
+#include "util.h"
+#include "utiltime.h"
+
+static int64_t PoolmanLowWater, PoolmanHighWater;
+static bool janitorInit = false;
+static bool floatingFee = false;
+static CFeeRate origTxFee;
+
+void TxMempoolJanitor()
+{
+    int64_t nStart = GetTimeMillis();
+
+    // remember original minimum tx fee
+    if (!janitorInit) {
+        janitorInit = true;
+        origTxFee = ::minRelayTxFee;
+    }
+
+    // query mempool stats
+    unsigned long totalTx = mempool.size();
+    int64_t totalSize = mempool.GetTotalTxSize();
```

Comment:
Can the mempool be passed as a parameter to TxMempoolJanitor() instead of using main.o's global?